pkg/controller/user-cluster-controller-manager/application-installation-controller/controller.go

```diff
@@ -0,0 +1,149 @@
+	appDefInformer, err := seedMgr.GetCache().GetInformer(ctx, &appkubermaticv1.ApplicationDefinition{})
+	if err != nil {
+		return fmt.Errorf("failed to get informer for applicationDefinition: %w", err)
+	}
+
+	if err = c.Watch(&source.Informer{Informer: appDefInformer}, handler.EnqueueRequestsFromMapFunc(enqueueAppInstallationForAppDef(ctx, log, r.userClient))); err != nil {
+		return fmt.Errorf("failed to watch applicationDefinition: %w", err)
+	}
+
+	return nil
+}
+
+// Reconcile ApplicationInstallation (ie install / update or uninstall applicationinto user cluster).
+func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (reconcile.Result, error) {
+	paused, err := r.clusterIsPaused(ctx)
+	if err != nil {
+		return reconcile.Result{}, fmt.Errorf("failed to check cluster pause status: %w", err)
+	}
+	if paused {
+		return reconcile.Result{}, nil
+	}
+
+	log := r.log.With("resource", request.Name)
+	log.Debug("Processing")
+
+	appInstallation := &appkubermaticv1.ApplicationInstallation{}
+
+	if err := r.userClient.Get(ctx, request.NamespacedName, appInstallation); err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Debug("applicationInstallation not found, returning")
+			return reconcile.Result{}, nil
+		}
+		return reconcile.Result{}, fmt.Errorf("failed to get applicationInstallation: %w", err)
+	}
+
+	err = r.reconcile(ctx, log, appInstallation)
+	if err != nil {
+		log.Errorw("ReconcilingError", zap.Error(err))
+		r.userRecorder.Event(appInstallation, corev1.EventTypeWarning, "ApplicationInstallationReconcileFailed", err.Error())
+	}
+
+	log.Debug("Processed")
+	return reconcile.Result{}, err
+}
+
+func (r *reconciler) reconcile(ctx context.Context, log *zap.SugaredLogger, appInstallation *appkubermaticv1.ApplicationInstallation) error {
+	// to implements logic in next PR
+	return nil
+}
+
+// enqueueAppInstallationForAppDef fan-out updates from applicationDefinition to the ApplicationInstallation that reference
+// this applicationDefinition.
+func enqueueAppInstallationForAppDef(ctx context.Context, log *zap.SugaredLogger, userClient ctrlruntimeclient.Client) func(object ctrlruntimeclient.Object) []reconcile.Request {
+	return func(applicationDefinition ctrlruntimeclient.Object) []reconcile.Request {
+		appList := &appkubermaticv1.ApplicationInstallationList{}
+		if err := userClient.List(ctx, appList); err != nil {
+			utilruntime.HandleError(fmt.Errorf("failed to list applicationInstallation: %w", err))
+			return []reconcile.Request{}
+		}
+
+		var res []reconcile.Request
+		for _, appInstallation := range appList.Items {
+			if appInstallation.Spec.ApplicationRef.Name == applicationDefinition.GetName() {
+				res = append(res, reconcile.Request{NamespacedName: types.NamespacedName{Name: appInstallation.Name}})
+			}
+		}
+
+		if res != nil {
+			log.Debugf("ApplicationDefinition '%s' has changed. Enqueue ApplicationInstallations %v", applicationDefinition.GetName(), res)
+		}
```

Comment:
I think this is redundant, given the logs in the Reconcile() function. This would be the only controller in KKP that logs which elements it puts on the queue.